src/components/implementation/no_interface/scheddev/sl.h

```diff
@@ -73,11 +73,29 @@ sl_thd_setprio(struct sl_thd *t, tcap_prio_t p)
 
 static inline struct sl_thd *
 sl_thd_lkup(thdid_t tid)
-{ return sl_mod_thd_get(sl_thd_lookup_backend(tid)); }
+{
+	assert(tid != 0);
+	if (tid > MAX_NUM_THREADS) {
+		return NULL;
+	}
+	return sl_mod_thd_get(sl_thd_lookup_backend(tid));
+}
+
+static inline thdid_t
+sl_thd_curr_id(void)
+{
+	thdid_t tid = cos_thdid();
+	assert(tid != 0);
+	assert(tid < MAX_NUM_THREADS);
+	return tid;
+}
```

Comment:
This safety checks would've saved me a ridiculous amount of time working on the CFE. I know we can't guard against every problem, but this has a good chance of saving someone time in the future.